### PR TITLE
Updated Cypress quickstarter dependencies

### LINKED ISSUE #441: Various warnings in E2E Cypress build
E.g.
```
npm WARN deprecated request@2.88.0: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated mkdirp@0.5.1: Legacy versions of mkdirp are no longer supported. Please update to mkdirp 1.x. (Note that the API surface has changed to use Promises in 1.x.)
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
```

```
tput: No value for $TERM and no -T specified
```

```
Browserslist: caniuse-lite is outdated. Please run the following command: `npm update`
```

The first build should have no warnings and use the latest and greatest if possible.

### PR DESCRIPTION
**Pull request by herrkoch**
(including major version update to Cypress 5)

Fixes #441 

Tasks: 
- [x] Updated documentation in `docs/modules/...` directory
- [x] Adjusted and verified tests in `tests/...` directory -> Tests ran fine in Jenkins of my OpenShift playground project

### CODE REVIEW

**Comment by herrkoch** [comment]
I just saw that there are a couple of warnings in the Jenkins build that I did not see locally (even though I removed `node_modules` completely). I'll try to reproduce and fix the warnings.

**Comment by herrkoch** [comment]
Ok one of the warnings about `tslint-language-service` you also mentioned in #440 @michaelsauter . It seems the plugin is deprecated in favor of https://github.com/microsoft/typescript-tslint-plugin . I'll try to replace it.

**Comment by herrkoch** [comment]
The other warning is about `rxjs-tslint`, which seems to provide obsolete tslint rules after the update to RxJS v6, so I am going to remove it.

**Comment by herrkoch** [comment]
The third and last deprecation warning is about `har-validator` which is a transitive dependency of Cypress, so unfortunately I can't do anything about it right now and hope the Cypress devs replace it soon. The PR is now ready for review.

File: e2e-cypress/files/tslint.json [hunk 1]
```diff
@@ -74,7 +71,6 @@
     "no-trailing-whitespace": true,
     "no-unnecessary-initializer": true,
     "no-unused-expression": true,
-    "no-use-before-declare": true,
```
Comment (herrkoch) [inline e2e-cypress/files/tslint.json:77]: Removed because of deprecation warning.

**Comment by herrkoch** [comment]
One last remark to this warning: `tput: No value for $TERM and no -T specified`-> That seems to come from the terminal configuration Jenkins uses, so not related to the Cypress quickstarter. I can't observe this locally.

**Comment by michaelsauter** [comment]
@herrkoch Awesome, looks good to me 👍 

Maybe @cschweikert or someone else approve to have some "frontend eyes" on this?

**Comment by netzartist** [comment]
Looks fine for me.

**Review by netzartist — APPROVED** [review]

**Pull request merged**